rust/windows-client/src/main.rs

```diff
@@ -0,0 +1,105 @@
+// Sample code from https://github.com/microsoft/windows-rs/blob/0.52.0/crates/samples/windows/create_window/src/main.rs
+
+use windows::{
+    core::*,
+    Win32::{
+        Foundation::*,
+        Graphics::Gdi::ValidateRect,
+        System::LibraryLoader::GetModuleHandleA,
+        UI::{Shell::*, WindowsAndMessaging::*},
+    },
+};
+
+fn main() -> Result<()> {
+    unsafe {
+        let instance = GetModuleHandleA(None)?;
+        debug_assert!(instance.0 != 0);
+
+        let window_class = s!("window");
+
+        let wc = WNDCLASSA {
+            hCursor: LoadCursorW(None, IDC_ARROW)?,
+            hInstance: instance.into(),
+            lpszClassName: window_class,
+
+            style: CS_HREDRAW | CS_VREDRAW,
+            lpfnWndProc: Some(wndproc),
+            ..Default::default()
+        };
+
+        let atom = RegisterClassA(&wc);
+        debug_assert!(atom != 0);
+
+        let window = CreateWindowExA(
+            WINDOW_EX_STYLE::default(),
+            window_class,
+            s!("This is a sample window"),
+            WS_OVERLAPPEDWINDOW | WS_VISIBLE,
+            CW_USEDEFAULT,
+            CW_USEDEFAULT,
+            CW_USEDEFAULT,
+            CW_USEDEFAULT,
+            None,
+            None,
+            instance,
+            None,
+        );
+
+        // Create tray icon
+
+        let sz_tip = {
+            let mut bytes = [0u8; 128];
+            let s = "Firezone client\0";
```

Comment:
I prefer not using `\0` in an &str since this can lead to some semantic confusion. We can use [`CString`](https://doc.rust-lang.org/std/ffi/struct.CString.html)  instead.